kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/DMNBaseConverter.java

```diff
@@ -74,6 +74,7 @@ protected void parseElements(HierarchicalStreamReader reader, UnmarshallingConte
                     null );
             if( object instanceof DMNModelInstrumentedBase ) {
                 ((DMNModelInstrumentedBase) object).setParent( (DMNModelInstrumentedBase) parent );
+                ( (DMNModelInstrumentedBase)parent).addChildren((DMNModelInstrumentedBase) object);
```

Comment:
Analogous of keeping a reference to the parent while deserializing, we also now keep a reference to the children Model nodes, so for easier validation (instead of walking the Model by reflection or with hard-coded visitors).